src/regdesk/registrant_details.js

```diff
@@ -0,0 +1,121 @@
+import { BadgeLabelBuilder } from './label_builder.js';
+
+/**
+ * Contains information relevant to a registration.
+ */
+export class RegistrantDetails {
+  /**
+   * Indicates whether this registrant has ever been checked in.
+   */
+  get hasBeenCheckedIn() {
+    return this.checkinDate !== undefined;
+  }
+
+  /**
+   * Indicates whether this registrant is under 18.
+   */
+  get isMinor() {
+    return this.age < 18;
+  }
+
+  /**
+   * Indicates whether the transaction was completed or not.
+   */
+  get isPaid() {
+    return this.paymentStatus === 'COMPLETED';
+  }
+
+  /**
+   * Initializes a new instance of the RegistrantDetails class
+   * @param {object} o - The argument object
+   * @param {string} o.preferredName - The registrant's preferred name.
+   * @param {string} o.legalName - The registrant's legal name.
+   * @param {string} o.birthdate - Birthday, in yyyy-MM-dd format.
+   * @param {string} o.amountDue - The remaining amount due.
+   * @param {string} o.badgeLine1 - The line 1 of the badge.
+   * @param {string} o.badgeLine2 - The line 2 of the badge.
+   * @param {string} o.sponsorLevel - The sponsorship level.
+   * @param {string} o.badgeId - The registrant ID to use as the badge ID.
+   * @param {string} o.conbookCount - The number of con books in the order.
+   * @param {string} o.checkinDate - The check in date string.
+   * @param {string} o.paymentStatus - The status string for the transaction.
+   */
```

Comment:
`numberOfBadgesPrintedSoFar` or similar a good idea?

Reply:
For our current purposes I both don't know that there's a really reliable way to pull this (the notes field is _complicated_) and I don't know that the actual quantity is useful. We're pulling in the checkinDate as our indicator that "this person has been checked in" and I think that's good enough for the moment.